cyder/base/utils.py

```diff
@@ -36,39 +36,39 @@ def shell_out(command, use_shlex=True):
     return out, err, p.returncode
 
 
-def log(msg, log_level='LOG_DEBUG', to_syslog=False, to_stderr=True,
-        logger=syslog):
-    msg = unicode(msg)
-    if to_syslog:
-        ll = getattr(logger, log_level)
-        for line in msg.splitlines():
-            logger.syslog(ll, line)
-    if to_stderr:
-        stderr.write(msg + '\n')
-
-
-def run_command(command, command_logger=None, failure_logger=None,
-                failure_msg=None, ignore_failure=False):
-    if command_logger:
-        command_logger('Calling `{0}` in {1}'.format(command, os.getcwd()))
+class Logger(object):
+    """
+    A Logger logs messages passed to it. Possible destinations include stderr
+    and the system journal (syslog).
+    """
 
-    out, err, returncode = shell_out(command)
+    def log_debug(self, msg):
+        pass
+
+    def log_info(self, msg):
+        pass
+
+    def log_notice(self, msg):
+        pass
 
+    def error(self, msg):
+        raise Exception(msg)
+
+
+def run_command(command, logger=Logger(), ignore_failure=False,
```

Comment:
I can't find anything concrete but I have a feeling that creating an instance in the arguments of a function is a bad practice. Normally you'd set the default argument to `logger=None` and then create the instance in the body of the function. In this case, I *think* this implementation works, because only one `Logger` instance is needed... but it might not be clear to someone reading the code whether that was your intention or if you didn't understand that this kind of thing could be hazardous.